cellarium/ml/core/datamodule.py

```diff
@@ -80,6 +80,15 @@ def __init__(
         num_workers: int = 0,
     ) -> None:
         super().__init__()
+
+        # We avoid saving the `DistributedAnnDataCollection` object to the hparams because that will save
+        # anndata objects in the model checkpoint.
+        # Instead, we save the class name and the init args which then can be used to
+        # re-initialize the distributed anndata.
+        # In order to achieve this, we temporarily re-assign `dadc` to its un-initialized state
+        # and then call `save_hyperparameters` which will save these values as hparams.
+        # Then, we re-assign `dadc` back to its initialized state.
+        # `initialize_object` handles the case when the object was passed as a dictionary of class path and init args.
         _dadc = dadc
         dadc = uninitialize_object(_dadc)
         self.save_hyperparameters(logger=False)
```

Comment:
can this be done more neatly with a context manager construct? also, `dadc` not not an attrib of self yet-- how would uninitializing it on line 93 have an effect on `self.save_hyperparameters` call on line 94?

Reply:
Having a context manager is good idea, let me think about it more.
`save_hyperparameters` looks up from the variable from the `locals()` using a argument name from the function signature.

Reply:
Another approach is to have `model` and `transforms` arg of `CellariumModule` as dicts of class name and init args and delegate their initialization to `CellariumModule` instead of `LightningCLI`. In fact, there is a `configure_model` method of `LightningModule` that is used to initialize large models with FSDP and DeepSpeed strategies (https://lightning.ai/docs/pytorch/stable/advanced/model_init.html#model-parallel-training-fsdp-and-deepspeed). We can just always use it to initialize all our models.